lib/Doctrine/ORM/Tools/SchemaValidator.php

```diff
@@ -406,6 +407,28 @@ function (array $fieldMapping) use ($class): ?string {
                             );
                         }
 
+                        if (
+                            isset($fieldMapping['enumType'])
+                            && $propertyType !== $fieldMapping['enumType']
+                            && interface_exists($propertyType)
+                            && is_a($fieldMapping['enumType'], $propertyType, true)
+                        ) {
+                            $backingType = (string) (new ReflectionEnum($fieldMapping['enumType']))->getBackingType();
+
+                            if ($metadataFieldType === $backingType) {
+                                return null;
+                            }
+
+                            return sprintf(
+                                "The field '%s#%s' has the metadata enumType '%s' with a backing type of '%s' that differs from the metadata field type '%s'.",
+                                $class->name,
+                                $fieldName,
+                                $fieldMapping['enumType'],
+                                $backingType,
+                                $metadataFieldType
+                            );
```

Comment:
Should this be extracted somehow, and reused on line 400? There could be the same improvement here (right?). Or maybe both conditions should somehow be combined? I having a hard time figuring out why you are not touching code already dealing with enums within this class :thinking:

Reply:
I'm having a hard time figuring out the same concern. I'm still checking, but they seem like different scenarios.

Reply:
Note that in this use case, the property type is an interface, so we don't really know what it's for. The only thing we can check is whether the mapped enum type is an implementation of that interface, and whether it matches the mapping type, nothing more.

Anyway, I will give it a try to see if by inverting the first block, I can merge something there.